native/effektio/src/api/messages.rs

```diff
@@ -33,6 +39,91 @@ impl RoomMessage {
     pub fn origin_server_ts(&self) -> u64 {
         self.inner.origin_server_ts.as_secs().into()
     }
+
+    pub fn msgtype(&self) -> String {
+        self.inner.content.msgtype().to_string()
+    }
+
+    pub async fn image_description(&self) -> Result<ImageDescription> {
+        match &self.inner.content.msgtype {
+            MessageType::Image(content) => {
+                let client = Client::current_client().unwrap();
+                let info = content.info.as_ref().unwrap();
+                RUNTIME.block_on(async move {
+                    let bin_data = client
+                        .get_media_content(
+                            &MediaRequest {
+                                source: content.source.clone(),
+                                format: MediaFormat::File,
+                            },
+                            false,
+                        )
+                        .await?;
+                    let description = ImageDescription {
+                        _bin_data: bin_data,
+                        _name: content.body.clone(),
+                        _mimetype: match info.mimetype.as_ref() {
+                            Some(value) => Some(value.clone()),
+                            None => None,
+                        },
+                        _size: match info.size {
+                            Some(value) => u64::from(value),
+                            None => 0,
+                        },
+                        _width: match info.width {
+                            Some(value) => Some(u64::from(value)),
+                            None => None,
+                        },
+                        _height: match info.height {
+                            Some(value) => Some(u64::from(value)),
+                            None => None,
+                        },
+                    };
+                    Ok(description)
+                })
+            }
+            _ => bail!("Invalid file format"),
+        }
+    }
+}
+
+pub struct ImageDescription {
+    _bin_data: Vec<u8>,
+    _name: String,
+    _mimetype: Option<String>,
+    _size: u64,
+    _width: Option<u64>,
+    _height: Option<u64>,
+}
+
+impl ImageDescription {
+    pub async fn bin_data(&self) -> Result<api::FfiBuffer<u8>> {
```

Comment:
this shouldn't copy here but only on request actually call to client and fetch the image and return it.

Reply:
Will move fetching function from message to client?